### ISSUE TITLE
New Python serverside implementation

### ISSUE THREAD

**Issue by alexcasalboni** [gmetais/sw-delta#4]
You can find it here:

https://github.com/alexcasalboni/sw-delta-python

Feedback is welcome :)

**Comment by gmetais** [comment]
This is huge! I'll look into it asap!

**Comment by alexcasalboni** [comment]
I will add tests, eventually!

Please notice that I used [diff-match-patch-python](https://github.com/JoshData/diff_match_patch-python), which returns a different diff format, wrt the nodejs version. I haven't tested its performance yet.

**Comment by gmetais** [comment]
The output is shorter with the given exemple, this is cool!

I don't know nothing about python so I fully trust you. I added it to the list of implementations.

We'll have to find a way to test the delta outputs cross languages. Any idea?

**Comment by alexcasalboni** [comment]
Yes, I guess the C++ implementation generates a shorter diff, in some cases.

I've just added some basic tests, for now.

In order to test the outputs cross-language, I'd go for a generic RESTful testing suit. It could take the server origin/basepath as configuration and perform integration tests btw client & server. Unit testing would still be delegated to each serverside implementation. Does it make sense?

**Comment by gmetais** [comment]
Tell me if I understand correctly:

There's a small http server, written in some language, let's say it's language A.

Then, any server-side language (languages A, B, C and D) runs delta encoding and sends the result to the small http server, that responds with the full file unencoded, so that the server-side implementation can check against the initial file.

**Comment by alexcasalboni** [comment]
I meant something different, actually.

There could be a test suit - written in any language (most likely JS) - that will invoke a given serverside implementations (A, B, C, or D) based on a configuration file. Eventually even all of them.
This requires that the given serverside implementation is running somewhere (even locally). This is more like an integration test, to verify that the given implementation works as expected from the client point of view.
